src/fs.c

```diff
@@ -0,0 +1,144 @@
+#include "system.h"
+
+#include "fs.h"
+#include "names.h"
+#include "xalloc.h"
+
+static bool makedir(const char *path, mode_t mode) {
+	if(!path) {
+		fprintf(stderr, "Trying to create directory, but path is not initialized");
+		fflush(stderr);
+		abort();
+	}
+
+	if(mkdir(path, mode) && errno != EEXIST) {
+		fprintf(stderr, "Could not create directory %s: %s\n", path, strerror(errno));
+		return false;
+	}
+
+	return true;
+}
+
+bool makedirs(tinc_dir_t dirs) {
+#define subdir(sub) snprintf(path, sizeof(path), "%s" SLASH sub, confbase)
```

Comment:
I'd rather avoid macros.

We do call `chdir(confbase)` very early when tinc starts up, so perhaps we can just simply do `makedir("sub")`.

Reply:
Most calls to this function are done by tincctl, not tincd. The macro actually doesn't save that much, let's remove it.